pkg/perf/perf.go

```diff
@@ -139,20 +167,45 @@ func (p *cache) MapForPID(pid int) (*Map, error) {
 		nsPid = p.nsPID[pid]
 	}
 
-	perfFile := fmt.Sprintf("/proc/%d/root/tmp/perf-%d.map", pid, nsPid)
-	h, err := hash.File(p.fs, perfFile)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, ErrPerfMapNotFound
+	fileFmts := []string{
+		"/proc/%d/root/tmp/perf-%d.map",
+		"/proc/%d/root/tmp/jit-%d.dump",
+		"/proc/%d/cwd/jit-%d.dump",
+	}
+
+	var perfFile string
+	var h uint64
+	for _, f := range fileFmts {
+		perfFile = fmt.Sprintf(f, pid, nsPid)
+		var err error
+		h, err = hash.File(p.fs, perfFile)
+		if err != nil && !os.IsNotExist(err) {
+			return nil, err
+		}
+		if h != 0 {
+			break
 		}
-		return nil, err
+	}
+
+	if h == 0 {
+		return nil, ErrPerfMapNotFound
 	}
 
 	if p.pidMapHash[pid] == h {
 		return p.cache[pid], nil
 	}
 
-	m, err := ReadMap(p.fs, perfFile)
+	var m Map
+	var err error
+	switch {
+	case strings.HasSuffix(perfFile, ".map"):
```

Comment:
How about converting the `Map` to an interface type and have 2 different implementations for `.map` and `.dump`?

Reply:
`Lookup(addr uint64) (string, error)` can be the only method.

Reply:
Sounds good, the idea went through my mind, I wired things up quickly to get it to work

Reply:
Actually, the interface feels like over-engineering too me, `MapFromDump()` would still be needed to compute end addresses and sort them to allow for binary search. And we would still have a switch-case to instantiate one or the other

Reply:
I don't think it's over-engineering. It will make this cleaner and easier to read. We write for humans to read and computers to run 😄

Reply:
hmm my thought exactly, simplicity is good for humans, the pattern would feel a bit forced to me :slightly_smiling_face:

Anyway I'd like to keep changes to the minimum in this file, we can revisit in another PR, I will only fixed the performance issues I introduced